docs/search.md

```diff
@@ -37,6 +37,619 @@ curl -XGET 'https://…/api/v1/search' -d '{
+`1 until n` and the input has to be provided by the user. 
+
+`to` :- This takes in the value as a timestamp according to ISO 8601 standards when the creation of ERC was ended.Current date and time can be entered as a value in this case.The cardinality for `to` is `1 until n` and the input has to be provided by the user. Both parameters can be used multiple times by providing the inputs in a sequential order.
+
+The values from the parameters are used in a pre-configured Elasticsearch [range query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html) which queries the database for defined duration
+
+```json
+{
+    "range": {
+        "metadata.o2r.temporal.begin": {
+             "from": from
+         }
+     }
+},
+{
+    "range": {
+        "metadata.o2r.temporal.end": {
+             "to": to
+        }
+     }
+}
+```
+`metadata.o2r.temporal.begin` and `metadata.o2r.temporal.end` are the fields within the dataset which will be queried for the duration which is set by the URL parameters.
+
+#### Parameters for Spatial Searching 
+
+`coords` :- This should cater for the coordinates defined by the bounding box. A 2-d array of coordinates is expected as a parameter. The coordinates are to be pairs of latitude longitude values describing a closed polygon.
+The value from this parameter is used in a pre-configured Elasticsearch [spatial query](https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-queries.html) which applies a `geo_shape` filter to fetch the desired ERC according to spatial parameters.The cardinality for `coords` is `1 until n` and the input has to be provided by the user. The parameter can be used multiple times to perform searching in different regions.
+
+```json
+{
+    "bool": {
+        "must": {
+            "match_all": {}
+         },
+         "filter": {
+              "geo_shape": {
+                   "metadata.o2r.spatial.geometry": {
+                        "shape": {
+                            "type": "polygon",
+                            "coordinates": coords
+                         },
+                         "relation": "within"
+                    }
+               }
+          }
+     }
+}
+```
+
+The filter has been nested within a `boolean/must match` query so that it completely matches the documents being queried. The filter has been applied to the `metadata.o2r.spatial.geometry` field of the dataset with a `within` relation so that the ERC that matches the document's coordinates completely are fetched.
+
+#### Parameters for Keyword Searching
+
+`keyword` :- Any word, phrase in the form of text that should be used to match the title of ERC.
+A simple match query is used at the back-end to match the query value with the values in dataset.The cardinality for `coords` is `n` and the input has to be provided by the user. The parameter can be used multiple times to perform searching.
```

Comment:
> the input has to be provided by the user.
Does this belong into the documentation?

Reply:
No, can be removed.